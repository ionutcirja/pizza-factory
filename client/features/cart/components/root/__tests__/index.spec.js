import React from 'react';
import { shallow } from 'enzyme';
import { Cart } from '..';

describe('Cart component', () => {
  let propsToRender;
  
  beforeEach(() => {
    propsToRender = {
      list: ['1', '2', '3'],
      theme: {
        colours: {
          red: 'red',
          white: 'white',
          darkBlue: 'dark blue',
        },
      },
    };
  });
  
  describe('render', () => {
    it('should render a shopping cart list, a place order button and a continue shopping button'
      + ' if list prop length is bigger than 0', () => {
      const wrapper = shallow(<Cart {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render a go to shopping button and a message'
      + ' if list prop length is equal with 0', () => {
      propsToRender.list = [];
      const wrapper = shallow(<Cart {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
