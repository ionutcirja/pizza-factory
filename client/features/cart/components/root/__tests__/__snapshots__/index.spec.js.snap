// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component render should render a go to shopping button and a message if list prop length is equal with 0 1`] = `
<CartWrapper>
  <Message
    colour="red"
  >
    You didn't add anything yet in your shopping bag.
  </Message>
  <ButtonsContainer>
    <Link
      replace={false}
      to="/"
    >
      <Button
        bgColour="dark blue"
        colour="white"
        type="button"
      >
        Go to factory
      </Button>
    </Link>
  </ButtonsContainer>
</CartWrapper>
`;

exports[`Cart component render should render a shopping cart list, a place order button and a continue shopping button if list prop length is bigger than 0 1`] = `
<CartWrapper>
  <List
    list={
      Array [
        "1",
        "2",
        "3",
      ]
    }
  />
  <Connect(WithTheme(Total)) />
  <ButtonsContainer>
    <Button
      bgColour="red"
      colour="white"
      type="button"
    >
      Place order
    </Button>
    <Link
      replace={false}
      to="/"
    >
      <Button
        bgColour="dark blue"
        colour="white"
        type="button"
      >
        Continue shopping
      </Button>
    </Link>
  </ButtonsContainer>
</CartWrapper>
`;
